File: examples/hello_go/hello.go
package main

var i int

func main() {
	i = 0
	hello()
}
func hello() {
	print("hello ")
	world()
}
func world() {
	print("world")
	if do(i) > 3 {
		more()
	}
}
func do(value int) int {
	print("!\n")
	return value + 5
}
func more() {
	i = 420
	stuff()
}
func stuff() {
	print("i is ", i, "\n")
}

File: examples/hello_c/hello.c

#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

int global = 1234;
int max = INT_MAX;
int min = INT_MIN;


int foo() {
    printf("bye!\n");
    exit(0);
}

int main() {
    int i = 1;
    printf("hello world!\n");
    i += 2;
    printf("i is %d\n", i);
    global = 4567;
    foo();
}
